### PR TITLE
feat: add withdrawal request report types and stubs

### PR DESCRIPTION
**Pull request by djordon**
## Description

This is on the path to https://github.com/stacks-network/sbtc/issues/798. Since we are skipping withdrawals for now, we stub out the required logic here.

## Changes

* Add types and functions for getting a withdrawal request report. Do not implement it though.
* Sort withdrawal requests by their qualified request ID.

## Testing Information

This PR doesn't add any tests, since there isn't much to test

## Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: signer/src/bitcoin/validation.rs [hunk 1]
```diff
@@ -360,6 +386,79 @@ impl DepositRequestReport {
...
+    pub request_id: u64,
+    /// The stacks transaction ID that lead to the creation of the
+    /// withdrawal request.
+    pub txid: StacksTxId,
+    /// Stacks block ID of the block that includes the transaction
+    /// associated with this withdrawal request.
+    pub block_hash: StacksBlockHash,
+    /// The amount of BTC, in sats, to withdraw.
+    pub amount: u64,
+    /// The max fee amount to use for the sBTC deposit transaction.
```
Comment (Jiloc) [inline signer/src/bitcoin/validation.rs:433]: typo: deposit -> withdrawal. 

Should we also add a hint that it's in sats like for the `amount` field?

File: signer/src/bitcoin/validation.rs [hunk 2]
```diff
@@ -360,6 +386,79 @@ impl DepositRequestReport {
...
+
+impl WithdrawalRequestReport {
+    fn qualified_id(&self) -> QualifiedRequestId {
+        QualifiedRequestId {
+            block_hash: self.block_hash,
+            txid: self.txid,
+            request_id: self.request_id,
+        }
+    }
+    /// Validate that the deposit request is okay given the report.
```
Comment (Jiloc) [inline signer/src/bitcoin/validation.rs:447]:
```suggestion
    /// Validate that the withdrawal request is okay given the report.
```

File: signer/src/bitcoin/validation.rs [hunk 3]
```diff
@@ -360,6 +386,79 @@ impl DepositRequestReport {
...
+    pub fn validate(&self, _: u64) -> Result<(), BitcoinWithdrawalOutputError> {
+        Err(BitcoinWithdrawalOutputError::Unknown(self.qualified_id()))
+    }
+
+    /// Validate that the fees assessed to the deposit prevout is below the
```
Comment (Jiloc) [inline signer/src/bitcoin/validation.rs:452]:
```suggestion
    /// Validate that the fees assessed to the withdrawal prevout is below the
```

**Review by Jiloc — COMMENTED** [review]
Just a few typos, everything else look good to me!

File: signer/src/bitcoin/validation.rs (hunk 1 above)
Comment (djordon) [reply signer/src/bitcoin/validation.rs]: fixed up in https://github.com/stacks-network/sbtc/pull/841/commits/4d70997616aa67eb9821bba3de938f3f7e8bc35e

File: signer/src/bitcoin/validation.rs (hunk 2 above)
Comment (djordon) [reply signer/src/bitcoin/validation.rs]: fixed up in https://github.com/stacks-network/sbtc/pull/841/commits/4d70997616aa67eb9821bba3de938f3f7e8bc35e

File: signer/src/bitcoin/validation.rs (hunk 3 above)
Comment (djordon) [reply signer/src/bitcoin/validation.rs]: https://github.com/stacks-network/sbtc/pull/841/commits/4d70997616aa67eb9821bba3de938f3f7e8bc35e

**Review by Jiloc — APPROVED** [review]
All good!

**Pull request merged**